src/libraries/Microsoft.PowerFx.Interpreter/Functions/LibraryMath.cs

```diff
@@ -552,92 +665,67 @@ public static FormulaValue Round(IRContext irContext, NumberValue[] args)
             var digitsArg = args[1].Value;
 
             var x = Round(numberArg, digitsArg);
+            if (x == double.NaN)
+            {
+                return CommonErrors.NumericOutOfRange(irContext);
+            }
+
```

Comment:
We should be able to have the numeric boundary checking for all functions in the 'StandardErrorHandling' wrapper, so it doesn't need to be done for each individual function

Reply:
I moved this to FiniteChecker method